File: codegen/results/v002/m3-max-48gb/ollama/qwen3.8_27b-mlx_nothink/20260910_071518/typescript/003_fibonacci/solution.ts
const N = parseInt(process.argv[2], 10);

if (isNaN(N) || N < 1) {
    // print nothing
    process.exit(0);
}

// Start with 1, 1
console.log(1);

if (N === 1) {
    process.exit(0);
}

console.log(1);

let a = 1;
let b = 1;

while (true) {
    const next = a + b;
    if (next > N) {
        break;
    }
    console.log(next);
    a = b;
    b = next;
}

File: codegen/results/v002/m3-max-48gb/ollama/qwen3.8_27b-mlx_nothink/20260910_071518/python/003_fibonacci/solution.py
import sys

def main():
    import sys
    args = sys.argv[1:]
    if not args:
        return
    try:
        N = int(args[0])
    except ValueError:
        return
    
    if N < 1:
        return
    
    # Start with 1, 1
    a, b = 1, 1
    while a <= N:
        print(a)
        a, b = b, a + b

if __name__ == "__main__":
    main()